base/math.jl

```diff
@@ -845,6 +845,17 @@ function _hypot(x::NTuple{N,<:Number}) where {N}
     end
 end
 
+function _hypot(x::NTuple{N,T}) where {N,T<:IEEEFloat}
+    infT = convert(T, Inf)
+    x = abs.(x) # doesn't change result but enables computational shortcuts
+    # note: any() was causing this to not inline for N=3 but mapreduce() was not
+    mapreduce(==(infT), |, x) && return infT # return Inf even if an argument is NaN
+    maxabs = reinterpret(T, maximum(z -> reinterpret(Signed, z), x)) # for abs(::IEEEFloat) values, a ::BitInteger cast does not change the result
+    iszero(maxabs) && return maxabs
+    scale,invscale = scaleinv(maxabs)
+    return scale * sqrt(sum(y -> abs2(y * invscale), x))
```

Comment:
> without the @fastmath that was causing build problems.

Since this method is for `IEEEFloat` only, I think we can create a local hack within `_hypot`?
```suggestion
    add_fast(x, y) = Core.Intrinsic.add_float_fast(x, y)
    return scale * sqrt(mapreduce(y -> abs2(y * invscale), add_fast, x))
```

Reply:
It'd probably be better to add a comment to explain why you're doing that.

Reply:
Done. I realized the `iszero` check above it was not necessary since the switch to use `scaleinv`. However, a NaN check is necessary so the `nonan` fastmath flag can't poison the result. So it still checks for 0 since that's basically free.